### PR TITLE
fix: rework contract handling in tests

### PR DESCRIPTION
**Pull request by daeMOn63**
- Remove inheritance between contracts & test cases (so we can use
  multiple contracts in a single test if we need to)
- Update default gas settings in local fetchd node to match mainnet
  (3000000 instead of 2000000)
- Add dataclass_json to wrap contract instantiate message in python class
- Move temp folder `.contract` inside project tree (currently writing
  outside of project tree)

### CODE REVIEW

File: scripts/00_setup_fetchd_local.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ sed -i 's/cors_allowed_origins = \[\]/cors_allowed_origins = \["\*"\]/' ~/.fetch
 
 # update the block parameters to match mainnet
 cp ~/.fetchd/config/genesis.json ~/.fetchd/config.genesis.json.bak
-jq '.consensus_params.block.max_gas = "2000000" | .consensus_params.block.max_bytes = "200000" | .consensus_params.evidence.max_bytes = "200000"' ~/.fetchd/config.genesis.json.bak > ~/.fetchd/config/genesis.json
+jq '.consensus_params.block.max_gas = "3000000" | .consensus_params.block.max_bytes = "300000" | .consensus_params.evidence.max_bytes = "300000"' ~/.fetchd/config.genesis.json.bak > ~/.fetchd/config/genesis.json
```
Comment (bryanchriswhite) [inline scripts/00_setup_fetchd_local.sh:25]: Where are you getting these values from?

Comment (daeMOn63) [reply scripts/00_setup_fetchd_local.sh:25]: you can check: 
- `fetchd query params subspace baseapp BlockParams`
- https://github.com/fetchai/FIPs/pull/5
- https://www.mintscan.io/fetchai/proposals/8

Comment (bryanchriswhite) [reply scripts/00_setup_fetchd_local.sh:25]: Thank you!

**Review by bryanchriswhite — APPROVED** [review]
Nice one @daeMOn63, thanks!

**Pull request merged**